packages/suite/src/actions/wallet/coinjoinClientActions.ts

```diff
@@ -257,38 +267,48 @@ export const onCoinjoinRoundChanged =
                 await dispatch(setBusyScreen(accountKeys));
                 dispatch(closeModal());
 
-                const completedSessions = coinjoinAccountsWithSession.filter(
-                    ({ session }) => session?.signedRounds?.length === session?.maxRounds,
+                const accountsWithSession = coinjoinAccountsWithSession.map(({ key }) =>
+                    selectAccountByKey(state, key),
                 );
 
-                if (completedSessions.length > 0) {
-                    const moreRoundsNeeded = completedSessions.some(({ key, targetAnonymity }) => {
-                        const account = selectAccountByKey(state, key);
-
-                        const { notAnonymized } = breakdownCoinjoinBalance({
-                            anonymitySet: account?.addresses?.anonymitySet,
-                            targetAnonymity,
-                            utxos: account?.utxo,
-                        });
+                // Simultaneously update all related accounts so that we have current data to check if any account has reached target anonymity.
+                await Promise.all(
+                    accountsWithSession.map(async account => {
+                        if (account) {
+                            await dispatch(coinjoinAccountActions.fetchAndUpdateAccount(account));
+                        }
+                    }),
+                );
 
-                        return !isZero(notAnonymized);
-                    });
+                const [accountsReachingAnonymity, accountsNotReachingAnonymity] = arrayPartition(
+                    coinjoinAccountsWithSession,
+                    ({ key }) => selectAreUtxosTooSmallByAccountKey(getState(), key),
```

Comment:
please rename the selector

Reply:
65a560cac224bb9a6375571ac3b08c02b6310d62